restapi/datasource_api_object.go

```diff
@@ -62,35 +67,46 @@ func dataSourceRestApiRead(d *schema.ResourceData, meta interface{}) error {
     path,
     path,
     path,
-    d.Id(),
+    "0", // We temporarily set ID of the new object to 0
     "{}",
     d.Get("debug").(bool),
   )
 
   if err != nil { return err }
-  log.Printf("resource_api_object.go: Data routine called. Object built:\n%s\n", obj.toString())
+  log.Printf("datasource_api_object.go: Data routine called. Object built:\n%s\n", obj.toString())
 
+  single_obj   := d.Get("single_object").(bool)
   search_key   := d.Get("search_key").(string)
   search_val   := d.Get("search_value").(string)
   results_key  := d.Get("results_key").(string)
   id_attribute := obj.api_client.id_attribute
   id := ""
   var data_array []interface{}
+  var single_object_data map[string]interface{}
 
   /*
     Issue a GET to the base path and expect results to come back
   */
-  res_str, err := obj.api_client.send_request("GET", path, "")
+  res_headers, res_body, err := obj.api_client.send_request("GET", path, "")
```

Comment:
Were you able to get this to compile? The current signature in `master` for send_request is `func (client *api_client) send_request (method string, path string, data string) (string, error)`... it returns only two objects, but this requires three.

Also, there are already areas in `send_request` that can dump headers and body - I'd prefer to keep that logic there.

Reply:
Ok.  I fixed this and tested that I could compile against my branch in my fork that only has the datasource changes.  Should be good now.

Reply:
No problem, mate, it's why we do reviews :smile:

Thanks for the quick fix!